Reading IR models from buffer

Quantizs opened the pull request:
resolves #15578

### This pullrequest changes

Added implementations to load IR models from buffer

dkurt commented:
@Quantizs, hi!

Please change a base branch to 3.4 and choose a different name for feature branch:

```
git checkout 3.4
git checkout -b branch_name
```

dkurt commented on modules/dnn/src/dnn.cpp:
```diff
@@ -2875,6 +2875,87 @@ Net Net::readFromModelOptimizer(const String& xml, const String& bin)
+
+    cvNet.impl->skipInfEngineInit = true;
+    return cvNet;
+}
```
Please create a separate method which returns `cv::Net` from `reader`. Let's reduce code duplication.

marcellBan commented:
@dkurt hi,

We are working on the requested changes, but as the merged branch cannot be changed (just the base) should we open a new PR? (We kind of missed the branch naming guidelines during our first read trough.)

dkurt commented:
@marcellBan, it's more about general recommendation, not OpenCV only. If you add a commit to a local `master` branch, then you just cannot keep this branch aligned with a remote one. So it's better to create a separate branches for local changes but keep origin branches unchanged for robust `pull` procedures.

marcellBan commented:
@dkurt, I totally understand and agree with it as I've worked with git for a few years now but I was asking if we should open a new PR with a different branch or should we just use this one and solve any upcoming git issues on our end.

dkurt commented:
@marcellBan, If it's OK to modify the branch names `master` locally, please try this:

1) rename `master` to something temporal like `fix`
2) go to branch `3.4`
3) `git checkout -b master`
4) `git cherry-pick your_commit_hash` or just copy-paste the code and create a new commit
5) push resulting branch with `-f` to the fork
6) reopen this PR. Click `Edit` button where you can change target branch to `3.4`

Quantizs commented:
We rebased the forked branch on 3.4 and I have refactored the code as requested and pushed it to `Quantizs:master` but i can't reopen the pull request or edit it.

dkurt commented:
@Quantizs, can you please try to open a new one?

Quantizs commented:
@dkurt Reopen under #16034
